### PR TITLE
[WIP] apollo-explorer

### PR DESCRIPTION
**Pull request by flyboarder**
## Hey, I just made a Pull Request!

I added an alternative to GraphiQL -> Apollo Studio Explorer

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: build/config.gypi [hunk 1]
```diff
@@ -0,0 +1,361 @@
+# Do not edit. File was generated by node-gyp's "configure" step
```
Comment (smyrick) [inline build/config.gypi:1]: Is this file supposed to be checked into the repo? I would think everything in the `/build`  directory would be excluded

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -4,55 +4,56 @@
...
+/docs/features/search                             @backstage/reviewers @backstage/techdocs-core
+/docs/features/techdocs                           @backstage/reviewers @backstage/techdocs-core
+/packages/search-common                           @backstage/reviewers @backstage/techdocs-core
+/packages/techdocs-cli                            @backstage/reviewers @backstage/techdocs-core
+/packages/techdocs-cli-embedded-app               @backstage/reviewers @backstage/techdocs-core
+/packages/techdocs-common                         @backstage/reviewers @backstage/techdocs-core
+/plugins/allure                                   @backstage/reviewers @deepak-bhardwaj-ps
+/plugins/apache-airflow                           @backstage/reviewers @cmpadden
+/plugins/api-docs                                 @backstage/reviewers @backstage/sda-se-reviewers
+/plugins/apollo-explorer                          @apollographql/solutions
```
Comment (Rugvip) [inline .github/CODEOWNERS:23]: This doesn't work afaik, teams need to exist within this org. Would need to list out the users instead. Please include `@backstage/reviewers` as well

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -4,55 +4,56 @@
...
+/plugins/sonarqube                                @backstage/reviewers @backstage/sda-se-reviewers
+/plugins/stack-overflow                           @backstage/reviewers @backstage/techdocs-core
+/plugins/stack-overflow-backend                   @backstage/reviewers @backstage/techdocs-core
+/plugins/techdocs                                 @backstage/reviewers @backstage/techdocs-core
+/plugins/techdocs-*                               @backstage/reviewers @backstage/techdocs-core
+/tech-insights-backend                            @backstage/reviewers @xantier @iain-b
+/tech-insights-backend-module-jsonfc              @backstage/reviewers @xantier @iain-b
+/tech-insights-tech-insights-common               @backstage/reviewers @xantier @iain-b
+/tech-insights-tech-insights-node                 @backstage/reviewers @xantier @iain-b
+yarn.lock                                         @backstage/reviewers @backstage-service
```
Comment (Rugvip) [inline .github/CODEOWNERS:59]: Looks like a duplicate down here. Also not sure why this file is displayed as completely changed, is it possible to keep the formatting similar enough for it to show the local diff?

File: plugins/apollo-explorer/package.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "@backstage/plugin-apollo-explorer",
+  "version": "0.1.0",
```
Comment (Rugvip) [inline plugins/apollo-explorer/package.json:3]: For new packages we start out at version `0.0.0` and then add a changeset with a `minor` bump. The changeset itself can give a brief introduction to the new package.

File: plugins/apollo-explorer/src/components/EmbeddedExplorer/EmbeddedExplorer.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          }`,
+          variables: {
+            test: 'abcxyz',
+          },
+          displayOptions: {
+            showHeadersAndEnvVars: true,
+          },
+        }}
+      />
+      {/* <ContentHeader title="Plugin title">
```
Comment (Rugvip) [inline plugins/apollo-explorer/src/components/EmbeddedExplorer/EmbeddedExplorer.tsx:57]: Please remove dead code

File: plugins/apollo-explorer/src/components/EmbeddedExplorer/EmbeddedExplorer.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      {' '}
+      {/* subtitle="Optional subtitle"> */}
+      <HeaderLabel label="Owner" value="Apollo GraphQL" />
+      <HeaderLabel label="Lifecycle" value="Alpha" />
+    </Header>
+    <Content noPadding stretch>
+      <ApolloExplorerReact
+        className="embeddableExplorer"
+        graphRef="acephei@current"
+        endpointUrl="https://acephei-gateway.herokuapp.com"
```
Comment (Rugvip) [inline plugins/apollo-explorer/src/components/EmbeddedExplorer/EmbeddedExplorer.tsx:42]: This doesn't seem to be accessible, could we use a publicly accessible example API instead?

Did you have a look at how the GraphiQL plugin sets up the endpoints as well btw? Feeling that it's quite important to have configurable endpoints with authentication for internal use-cases.

File: plugins/apollo-explorer/src/components/EmbeddedExplorer/EmbeddedExplorer.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  <Page themeId="tool">
+    <Header title="Apollo Studio Explorer">
+      {' '}
+      {/* subtitle="Optional subtitle"> */}
+      <HeaderLabel label="Owner" value="Apollo GraphQL" />
+      <HeaderLabel label="Lifecycle" value="Alpha" />
+    </Header>
+    <Content noPadding stretch>
+      <ApolloExplorerReact
+        className="embeddableExplorer"
```
Comment (Rugvip) [inline plugins/apollo-explorer/src/components/EmbeddedExplorer/EmbeddedExplorer.tsx:40]: Please avoid global css styling, use `makeStyles` from MUI for example instead.

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Couple of cleanup opportunities :grin:

**Comment by flyboarder** [comment]
Thanks for the helpful comments, I'll get those cleaned up.

**Pull request closed without merging**